src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -34,8 +34,27 @@ const cytscapeGraphStyle = style({
   left: 220
 });
 
+const makeCancelablePromise = (promise: Promise<any>) => {
```

Comment:
Can you check if there is already a lib to do this? If so, please add as a follow up (or write a jira) for that.
Others might benefit from this cancel-able promise pattern which is what is suggested for this kind of scenario.
Maybe we can also refactor so that instead of a Timeout is a cancel-able promise and we get rid of the pollTimeoutRef.

Reply:
IMO: Observables are a much cleaner approach when you start talking about canceling/merging/combining promises. Promises are one-off; Observables are streams that can be composed/combined/filtered. That said, I recommend [Redux-Observable](https://redux-observable.js.org/docs/basics/Epics.html) for future work.